sw/device/tests/dif/dif_otbn_unittest.cc

```diff
@@ -293,44 +293,46 @@ TEST_F(IsBusyTest, Success) {
   EXPECT_EQ(busy, true);
 }
 
-class GetErrCodeTest : public OtbnTest {};
+class GetErrBitsTest : public OtbnTest {};
 
-TEST_F(GetErrCodeTest, NullArgs) {
+TEST_F(GetErrBitsTest, NullArgs) {
   dif_otbn_result_t result;
-  dif_otbn_err_code_t err_code;
+  uint32_t err_bits;
 
-  result = dif_otbn_get_err_code(nullptr, nullptr);
+  result = dif_otbn_get_err_bits(nullptr, nullptr);
   EXPECT_EQ(result, kDifOtbnBadArg);
 
-  result = dif_otbn_get_err_code(&dif_otbn_, nullptr);
+  result = dif_otbn_get_err_bits(&dif_otbn_, nullptr);
   EXPECT_EQ(result, kDifOtbnBadArg);
 
-  err_code = kDifOtbnErrCodeBadDataAddr;
-  result = dif_otbn_get_err_code(nullptr, &err_code);
+  err_bits = (1 << kDifOtbnErrBitBadDataAddr);
+  result = dif_otbn_get_err_bits(nullptr, &err_bits);
   EXPECT_EQ(result, kDifOtbnBadArg);
-  EXPECT_EQ(err_code, kDifOtbnErrCodeBadDataAddr);
+  EXPECT_EQ(err_bits, (1 << kDifOtbnErrBitBadDataAddr));
```

Comment:
Seems odd to be checking that a BadArg result preserves the other argument?

Reply:
Well, if the function contract is something like "if you do something silly, I'll return with an error and won't trash anything behind the pointer you gave me", then this seems like a good idea. But maybe we should be setting `err_bits` to `0x1234` to make it clear that this is just an arbitrary canary value?

Reply:
Again I'm just preserving existing behaviour, I can drop the check if it's felt it's not needed, @imphil?

Reply:
I've left this as is given @imphil's comments about the function contract

Reply:
I think we should revisit being more consistent about how DIFs specify their contracts. There's definitely a pattern where some DIFs are way over-constrained, resulting in tests that are also *way* over-constrained. Can someone file an issue for OTBN?